### PR TITLE
Enable unicode normalization by default for speech

### LINKED ISSUE #16616: Unicode normalization: Enable by default for speech and reconsider character navigation
### Is your feature request related to a problem? Please describe.
Recently, Unicode normalization has been added to NVDA, but in #16466, there was some discussion about whether it should be enabled by default.
Furthermore, there was some debate about character navigation, i.e. whether normalization should be enabled by default for character navigation, and/or whether the delayed character description feature could be utilized to some extend.

### Describe the solution you'd like
1. Enable normalization by default for speech output.
2. When enabled, always normalize (i.e. also with character navigation) but prepend the word 'normalized' or something less technical to the character.

### Describe alternatives you've considered
@CyrilleB79 proposed a three choice option in https://github.com/nvaccess/nvda/issues/16466#issuecomment-2132149218:

1. Normalization disabled (Current default)
2. Normalization enabled to read text but disabled to read by character (current behavior of enabled)
3. Normalization enabled for all (not yet implemented)

I'm happy to consider this alternative as well, though it will probably take more effort. I'm now at the point where I think if we add a proper term prepended for character navigation, that would be the best solution.

### Additional context
1. I'm reluctant to take the character descriptions feature into account:

    * Current use of these is for phonetics. I'm inclined to say that adding Unicode character descriptions is a fundamental change to this feature
    * There are no translated names for the several mathematical Unicode chars in CLDR

2. Braille is beyond discussion.

### Discussion guidelines
When attending to the discussion, please avoid repeating standpoints over again that were already stated in https://github.com/nvaccess/nvda/issues/16466. It is clear that we have different opinions and that is okay.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #16616 

### Summary of the issue:
In #16521, Unicode normalization was added, but it is disabled by default.

### Description of user facing changes
Unicode normalization is now enabled by default for Speech.

### Description of development approach
Change default values

### Testing strategy:
Ensured that the default is now enabled for speech.

### Known issues with pull request:
Unicode normalization is still disabled by default for Braille. For speech, if someone defines symbol pronunciation for a character, it is never normalized, even when normalization is on. For Braille however, we are unable to find out pre-translation whether liblouis knows about a character. It is possible that an unnormalized character translates correctly, whereas the normalized equivalent does not.

The normalization may have unwanted effects in some situations, such as [the Portuguese abbreviation "nº" being misread by some synthesizers after normalization](https://github.com/nvaccess/nvda/issues/16616#issuecomment-2297700026), or making it difficult to tell the meaning of some mathematical symbols such as "ⁱ" (superscript i) or "ₙ" (subscript n).

### Code Review Checklist:

- [ ] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Unicode normalization is now enabled by default, enhancing text-to-speech output.
	- eSpeak NG upgraded to version 1.52-dev, adding support for new languages: Faroese and Xextan.
	- Improved support for multi-line braille displays.

- **Bug Fixes**
	- Command line options `-c`/`--config-path` and `--disable-addons` now respected during updates.

- **Documentation**
	- Updated user guide to reflect the new default setting for Unicode normalization. 

- **Chores**
	- Enhanced stability of NVDA's Poedit support, increasing the minimum required version to 3.5.

### CODE REVIEW

**Comment by seanbudd** [comment]
@LeonarddeR - we think that we should go ahead, and that normalization is generally preferred behaviour to that example. However, if cases like this begin to stack up, we can reconsider switching it back to disabled by default

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ The available options are:
 * The `-c`/`--config-path` and `--disable-addons` command line options are now respected when launching an update from within NVDA. (#16937)
 * eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
   * Added new languages Faroese and Xextan.
+* The default state of Unicode Normalization for speech output has been changed to enabled. (#17017, @LeonarddeR).
+  * You can still disable this functionality in the speech settings categorie in the NVDA Settings dialog.
```
Comment (seanbudd) [inline user_docs/en/changes.md:27]:
```suggestion
  * You can still disable this functionality in the speech settings category in the NVDA Settings dialog.
```

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ The available options are:
 * The `-c`/`--config-path` and `--disable-addons` command line options are now respected when launching an update from within NVDA. (#16937)
 * eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
   * Added new languages Faroese and Xextan.
+* The default state of Unicode Normalization for speech output has been changed to enabled. (#17017, @LeonarddeR).
```
Comment (seanbudd) [inline user_docs/en/changes.md:26]:
```suggestion
* Unicode Normalization is now enabled by default for speech output. (#17017, @LeonarddeR).
```

**Review by seanbudd — COMMENTED** [review]
Is this ready for review / merge?

File: user_docs/en/changes.md (hunk 1 above)
Comment (SaschaCowley) [reply user_docs/en/changes.md:27]:
```suggestion
  * You can still disable this functionality in the Speech category of the NVDA Settings dialog.
```

**Review by SaschaCowley — APPROVED** [review]

**Comment by CyrilleB79** [comment]
(https://github.com/nvaccess/nvda/issues/16616#issuecomment-2328021474) was ment to be here; copying it below:

Could we add at least in the "Known issues" of this PR what is described in https://github.com/nvaccess/nvda/issues/16616#issuecomment-2297700026, i.e. that the normalization may have unwanted effect in some situation / with some characters (e.g. "nº", common indices or exponents in math expressions such as "ⁱ" or "ₙ")

Moreover, I wonder if we could add a third option for normalization which would be "When suitable", forcing normalization in Word equations and other situations where it would be required and keeping it disabled in other cases. But that would be for another PR in any case.

**Comment by seanbudd** [comment]
I agree that it should be listed in the known issues section of the PR description. I think a third option for "when suitable" should be proposed in a separate issue

File: user_docs/en/changes.md [hunk 3]
```diff
@@ -89,8 +89,9 @@ There have also been a number of fixes, including to mouse tracking in Firefox,
 * The `-c`/`--config-path` and `--disable-addons` command line options are now respected when launching an update from within NVDA. (#16937)
 * eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
   * Added new languages Faroese and Xextan.
+* Unicode Normalization is now enabled by default for speech output. (#17017, @LeonarddeR).
+  * You can still disable this functionality in the Speech category of the NVDA Settings dialog.
```
Comment (seanbudd) [inline user_docs/en/changes.md:93]: this needs to be moved to 2025.1

**Comment by SaschaCowley** [comment]
@coderabbitai Please create an issue for adding a "when suitable" option to the unicode normalization setting of NVDA's speech options. Please remember to use the NVDA feature request template, and use the discussion in this PR to inform the reasoning in the issue you create.

File: user_docs/en/changes.md [hunk 4]
```diff
@@ -90,7 +92,6 @@ There have also been a number of fixes, including to mouse tracking in Firefox,
 * eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
   * Added new languages Faroese and Xextan.
 * When using a multi-line braille display via the standard HID braille driver, all lines of cells will be used. (#16993, @alexmoon)
-* The stability of NVDA's Poedit support has been improved with the side effect that the minimum required version of Poedit is now version 3.5. (#16889, @LeonarddeR)
```
Comment (seanbudd) [inline user_docs/en/changes.md:93]: please remove the change to 2024.4 change log

Comment (SaschaCowley) [reply user_docs/en/changes.md]: No idea how that got deleted, whoops.

**Review by seanbudd — APPROVED** [review]

**Review by Qchristensen — APPROVED** [review]
Just looking at the large Userguide here and that looks fine :)

**Comment by seanbudd** [comment]
@LeonarddeR - this PR seems to break system tests. Can you investigate

**Comment by LeonarddeR** [comment]
I think this is because `speech.speech.processText` uses `rstrip` instead of `strip`. `strip` is too aggressive for normalization though.
For now, I disabled normalization in the system tests. I don't think system tests have to test normalization, that's more something for unit tests. The way NVDA introduces spaces in speech sequences is really messy and should be investigated at some point, but that's out of scope for this pr and changing things will likely break other things.

**Comment by XLTechie** [comment]
@LeonarddeR
> The way NVDA introduces spaces in speech sequences is really messy and should be investigated at some point

Can you open an issue or discussion on that, so it can be tagged for someone to work on in the future, when you're not available to explain the problems you noticed with it?

I haven't looked at that in particular, so maybe it's obvious, but one person's "obvious" is often another's "what is he talking about?" :)

**Pull request merged**